code/modules/projectiles/guns/energy/kinetic_accelerator.dm

```diff
@@ -144,10 +144,10 @@
 	overheat = FALSE
 
 /obj/item/gun/energy/kinetic_accelerator/update_icon()
-	..()
-
 	if(empty_state && !can_shoot())
```

Comment:
If you set an empty state for everything you can forgo the check (further optimizing)